testing/test_config.py

```diff
@@ -583,3 +583,60 @@ def test_with_specific_inifile(self, tmpdir):
         inifile = tmpdir.ensure("pytest.ini")
         rootdir, inifile, inicfg = determine_setup(inifile, [tmpdir])
         assert rootdir == tmpdir
+
+class TestOverrideIniArgs:
+    """ test --override-ini """
+    @pytest.mark.parametrize("name", "setup.cfg tox.ini pytest.ini".split())
+    def test_override_ini_names(self, testdir, name):
+        testdir.tmpdir.join(name).write(py.std.textwrap.dedent("""
+            [pytest]
+            custom = 1.0
+        """))
+        testdir.makeconftest("""
+            def pytest_addoption(parser):
+                parser.addini("custom", "")
+        """)
+        testdir.makepyfile("""
+            import pytest
+            def test_pass():
+                ini_val = pytest.config.getini("custom")
+                print('\\ncustom_option:%s\\n' % ini_val)
```

Comment:
bad style because of using global state: better just accept ``pytestconfig`` as a fixture into the test.